flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/FlinkUtils.java

```diff
@@ -72,6 +72,12 @@ public static Configuration getEffectiveConfig(FlinkDeployment flinkApp) {
                         KubernetesConfigOptions.ServiceExposedType.ClusterIP);
             }
 
+            if (spec.getServiceAccount() != null) {
```

Comment:
Maybe we need to set `KubernetesConfigOptions#KUBERNETES_SERVICE_ACCOUNT` here. If the user wants to give the TaskManager pods less permissions, then they should create different service accounts for both and configure them in jobManager and taskManager fields.

Note: The reason why we need to configure service account for TaskManager pod is about the Kubernetes HA. TaskManagers retrieve leader information from K8s ConfigMap.

The e2e test is failing also because of this.

Reply:
Thanks for catching the config parameter issue. Fix is pushed. This just reinforces why we should not expose such implementation details to user (CR should abstract how we interact with k8s and also work for standalone mode).

Reply:
We also need to add the field `serviceAccount` to `JobManagerSpec` and `TaskManagerSpec`. Right? Otherwise, user could not configure service account for JobManager pod separately or have different service accounts for both.

Reply:
Since separate service accounts for jm and tm pods are a rare advanced usage, they can be covered with the pod templates, which are already available separately for jm and tm.

Reply:
Make sense to me.